src/Database/Database.php

```diff
@@ -120,6 +120,36 @@ class Database
      */
     static protected $filters = [];
 
+    /**
+     * @var Authorization
+     */
+    protected Authorization $read;
+
+    /**
+     * @var Authorization
+     */
+    protected Authorization $write;
+
+    /**
+     * Get read authorization
+     * 
+     * @return Authorization
+     */
+    public function getRead(): Authorization
```

Comment:
I don't think `$database->getRead()` tells the right story. Maybe:
```suggestion
    public function getReadValidator(): Authorization
```
I think we should be more verbose with these variables/methods, else the context gets lost.